br/pkg/encryption/master_key/mem_backend.go

```diff
@@ -0,0 +1,95 @@
+// Copyright 2024 PingCAP, Inc. Licensed under Apache-2.0.
+
+package encryption
+
+import (
+	"context"
+	"crypto/aes"
+	"crypto/cipher"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/kvproto/pkg/encryptionpb"
+	"github.com/pingcap/tidb/br/pkg/kms"
+)
+
+const (
+	gcmTagNotFound = "aes gcm tag not found"
+	wrongMasterKey = "wrong master key"
+)
+
+type MemAesGcmBackend struct {
+	key *kms.PlainKey
+}
+
+func NewMemAesGcmBackend(key []byte) (*MemAesGcmBackend, error) {
+	plainKey, err := kms.NewPlainKey(key, kms.CryptographyTypeAesGcm256)
+	if err != nil {
+		return nil, errors.Annotate(err, "failed to create new mem aes gcm backend")
+	}
+	return &MemAesGcmBackend{
+		key: plainKey,
+	}, nil
+}
+
+func (m *MemAesGcmBackend) EncryptContent(_ctx context.Context, plaintext []byte, iv IV) (*encryptionpb.EncryptedContent, error) {
+	content := encryptionpb.EncryptedContent{
+		Metadata: make(map[string][]byte),
+	}
+	content.Metadata[MetadataKeyMethod] = []byte(MetadataMethodAes256Gcm)
+	content.Metadata[MetadataKeyIv] = iv[:]
+
+	block, err := aes.NewCipher(m.key.Key())
+	if err != nil {
+		return nil, err
+	}
+	aesgcm, err := cipher.NewGCM(block)
+	if err != nil {
+		return nil, err
+	}
+
+	ciphertext := aesgcm.Seal(nil, iv[:], plaintext, nil)
+	content.Content = ciphertext[:len(ciphertext)-aesgcm.Overhead()]
+	content.Metadata[MetadataKeyAesGcmTag] = ciphertext[len(ciphertext)-aesgcm.Overhead():]
```

Comment:
nit: would you add a reference link about the format? Directly assuming that the first `Overhead()` bytes are the tag looks somehow magic.

Reply:
sounds good!